include/tables/DBTable.h

```diff
@@ -279,6 +280,34 @@ std::vector<TableRow> DBTable<TableRow>::doRead(Ndb* connection, std::string ind
   return results;
 }
 
+template<typename TableRow>
+int DBTable<TableRow>::deletesByIndex(Ndb* connection, std::string index, AnyMap& any, boost::optional<Int64> partitionId) {
+  start(connection, partitionId);
+  LOG_INFO(getName() << " -- deletesByIndex with index : " << index);
+  mIndex = getIndex(mDatabase, index);
+  NdbIndexScanOperation* operation = getNdbIndexScanOperation(mCurrentTransaction, mIndex);
+  operation->readTuples(NdbOperation::LM_Exclusive);
```

Comment:
Does hive use locks?

Reply:
It kinda. Hive deletes every thing in one transaction. Now that we delete these rows in multiple transactions and these triggers can be called in a circle it would be safe to take exclusive locks. Exclusive locks should not be a problem as these scan operation are not supposed to retun lots of rows.